### PR TITLE
Feature/jc19 504 item management react

### PR DESCRIPTION
**Pull request by LocTran-saritasa**

### CODE REVIEW

File: apps/react/src/features/anime-list/components/Anime/Anime.tsx [hunk 1]
```diff
@@ -14,40 +29,96 @@ interface Props {
...
+  const isDeleting = useAppSelector(selectIsEditingAnime);
+  const [openDelete, setOpenDelete] = useState(false);
+  const { navigateWithSearchParams } = useAppNavigate();
+
+  const handleDeleteAnime = (event: MouseEvent) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setOpenDelete(true);
+  };
+  const handleCloseDeleteAnime = () => setOpenDelete(false);
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/Anime/Anime.tsx:42]: Please add empty rows between `hadle...` functions

File: apps/react/src/features/anime-list/components/Anime/Anime.tsx [hunk 2]
```diff
@@ -14,40 +29,96 @@ interface Props {
...
-              variant="rounded"
-              alt=""
-              src={anime.image}
-            />
-          </ListItemAvatar>
-          <div className={styles['info__data']}>
-            <span className={styles['info__data--primary']}>
+    <>
+      <NavLink
+        to={`${anime.id}?${searchParams.toString()}`}
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/Anime/Anime.tsx:56]: Let's move path to const 

`const to = `${anime.id}?${searchParams.toString()}`;`

File: apps/react/src/features/anime-list/components/ManagementForm/ManagementForm.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  });
+
+  useEffect(() => {
+    dispatch(fetchGenres());
+  }, []);
+
+  const fetchGenreByKey = (key: string) => {
+    dispatch(fetchGenresByKey(key));
+  };
+  const addGenre = (name: string) => {
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/ManagementForm.tsx:109]: Please add empty rows between const

File: apps/react/src/features/anime-list/components/ManagementForm/components/FormChipsSelect/FormChipsSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  /** Is loading options or not. */
+  readonly isLoading?: boolean;
+
+  /** Add new option. */
+  readonly addOption: (name: string) => void;
+}
+
+/** Delay time while search option (ms). */
+const delayOnSearch = 300;
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/components/FormChipsSelect/FormChipsSelect.tsx:35]: https://wiki.saritasa.rocks/frontend/languages/ts-js/naming#naming-convention

PLease rename to `DELAY_ON_SEARCH`.

File: apps/react/src/features/anime-list/components/ManagementForm/components/FormChipsSelect/FormChipsSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  addOption,
+}: Props<T>) => {
+  const [search, setSearch] = useState<string>('');
+
+  useEffect(() => {
+    const searchOption = setTimeout(() => fetchByKey(search), delayOnSearch);
+    return () => clearTimeout(searchOption);
+  }, [search]);
+
+  const onSearchChange = (event: SyntheticEvent, value: string) => {
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/components/FormChipsSelect/FormChipsSelect.tsx:53]: It should be `handleSearchChange`. The prefix `on` good if you receive props, not for handling something.

File: apps/react/src/features/anime-list/components/ManagementForm/components/FormDateSelect/FormDateSelect.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  readonly initialValue: Date;
+
+  /** Label name. */
+  readonly label: string;
+
+  /** Field name. */
+  readonly name: string;
+
+  /** Handle date change. */
+  readonly handleDateChange: (date: Date) => void;
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/components/FormDateSelect/FormDateSelect.tsx:21]: This is `props`. It should be `onDateChange`.

File: apps/react/src/features/anime-list/components/ManagementForm/components/FormDateSelect/FormDateSelect.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+const FormDateSelectComponent: FC<Props> = ({
+  handleDateChange,
+  initialValue,
+  name,
+  label,
+}) => {
+  const [value, setValue] = useState<Date | null>(initialValue);
+
+  const onDateChange = (newValue: Date | null) => {
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/components/FormDateSelect/FormDateSelect.tsx:32]: It should be handleDateChange.

File: apps/react/src/features/anime-list/components/ManagementForm/components/ImageSelect/ImageSelect.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+import styles from './ImageSelect.module.css';
+
+interface Props {
+
+  /** Initial image. */
+  readonly initialImage?: string;
+
+  /** Change handler. */
+  readonly handleChange: (image: File) => void;
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/components/ImageSelect/ImageSelect.tsx:12]: `onChange`

File: apps/react/src/features/anime-list/components/ManagementForm/components/ImageSelect/ImageSelect.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    };
+    if (file) {
+      reader.readAsDataURL(file);
+      handleChange(file);
+    }
+  };
+  return (
+    <div className={styles['selector']}>
+      {imageUrl && (
+        <img className={styles['selector__image']} src={imageUrl} />
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/components/ImageSelect/ImageSelect.tsx:34]: Missed required `alt` attribute.

**Review by golovachev-yaroslav — CHANGES_REQUESTED** [review]

**Review by golovachev-yaroslav — APPROVED** [review]

File: apps/react/src/features/anime-list/components/ManagementForm/ManagementForm.tsx [hunk 10]
```diff
@@ -142,33 +112,25 @@ const ManagementFormComponent: FC<Props> = ({ managementData, onSubmit }) => {
           <FormDateSelect
             name="aired.start"
             label="Aired start"
-            initialValue={new Date(formik.getFieldMeta('aired.start').value)}
+            initialValue={
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/ManagementForm.tsx:115]: Please move to const, it makes more readable the code.

File: apps/react/src/features/anime-list/components/ManagementForm/ManagementForm.tsx [hunk 11]
```diff
@@ -142,33 +112,25 @@ const ManagementFormComponent: FC<Props> = ({ managementData, onSubmit }) => {
...
           <FormDateSelect
             name="aired.end"
             label="Aired end"
-            initialValue={new Date(formik.getFieldMeta('aired.end').value)}
+            initialValue={
+              formik.getFieldMeta('aired.end').value ?
+                new Date(formik.getFieldMeta('aired.end').value) :
+                null
+            }
             handleDateChange={handleEndDateChange}
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/ManagementForm.tsx:130]: It should be onDateChange, bacause it's props.

File: apps/react/src/features/anime-list/components/ManagementForm/components/ChipsEditor/ChipsEditor.tsx [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+const ChipsEditorComponent: FC = () => {
+  const genres = useAppSelector(selectGenres);
+  const studios = useAppSelector(selectStudios);
+  const isGenresLoading = useAppSelector(selectAreGenresLoading);
+  const isStudiosLoading = useAppSelector(selectAreStudiosLoading);
+  const dispatch = useAppDispatch();
+  const fetchGenreByKey = (key: string) => {
+    dispatch(fetchGenresByKey(key));
+  };
+  const addGenre = (name: string) => {
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/components/ChipsEditor/ChipsEditor.tsx:21]: Please add empty rows between handle functions.

File: apps/react/src/features/anime-list/components/ManagementForm/components/ChipsEditor/ChipsEditor.tsx [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+  return (
+    <>
+      <FormChipsSelect
+        label="Genre select"
+        name="genresData"
+        options={genres}
+        getOptionName={(genreOption: Genre) => genreOption.name}
+        isLoading={isGenresLoading}
+        fetchByKey={fetchGenreByKey}
+        addOption={addGenre}
```
Comment (golovachev-yaroslav) [inline apps/react/src/features/anime-list/components/ManagementForm/components/ChipsEditor/ChipsEditor.tsx:40]: Passed functions should starts with prefix on

**Review by golovachev-yaroslav — CHANGES_REQUESTED** [review]

**Review by golovachev-yaroslav — APPROVED** [review]

**Review by artem-sotnichenko-saritasa — APPROVED** [review]

**Review by golovachev-yaroslav — APPROVED** [review]

**Pull request closed without merging**